torch/csrc/jit/testing/file_check.cpp

```diff
@@ -293,6 +298,60 @@ struct FileCheckImpl {
     }
   }
 
+  // Checks that source token is highlighted, does not advance search range.
+  void doCheckSourceHighlighted(
+      const Check& check,
+      const std::shared_ptr<Source>& source,
+      size_t start_offset) {
+    auto pos = source->text().find(check.search_str_, start_offset);
+    if (pos == std::string::npos) {
+      // Guaranteed to fail to generate error message.
+      assertFind(source, check.search_str_, start_offset, check);
+      return;
+    }
+
+    auto lineno = source->lineno_for_offset(pos);
+    auto col = pos - source->offset_for_line(lineno);
+    auto highlight_lineno = lineno + 1;
+
+    auto construct_error_and_throw = [&]() {
+      SourceRange error_range(source, pos, check.search_str_.size());
+      std::stringstream ss;
+      ss << "Expected to find ";
+      c10::printQuotedString(ss, check.search_str_);
+      ss << "highlighted but it is not." << std::endl;
+      error_range.highlight(ss);
+      throw std::runtime_error(ss.str());
+    };
+
+    if (highlight_lineno >= source->num_lines()) {
+      construct_error_and_throw();
+    }
+
+    auto highlight_start_offset =
+        source->offset_for_line(highlight_lineno) + col;
+    auto highlight_end_offset = std::min(
+        highlight_start_offset + check.search_str_.size(),
+        source->text().size());
+
+    if (highlight_end_offset >= source->text().size()) {
```

Comment:
Can `highlight_end_offset > source->text().size()` ever be true? It's a small thing and doing `==` is probably less safe and harder to debug later so I would not change it.

Reply:
I think it can be true. Say a segment of text with a short last line.